lib/liquid/block_body.rb

```diff
@@ -88,34 +92,30 @@ def render(context)
           context.push_interrupt(node.interrupt)
           break
         else # Other non-Block tags
-          render_node_to_output(node, output, context)
+          render_node(context, output, node)
           break if context.interrupt? # might have happened through an include
         end
         idx += 1
+
+        raise_if_resource_limits_reached(context, output.bytesize - previous_output_size)
       end
 
-      output.join
+      output
     end
 
     private
 
-    def render_node_to_output(node, output, context, skip_output = false)
-      node_output = node.render(context)
-      node_output = node_output.is_a?(Array) ? node_output.join : node_output.to_s
-      check_resources(context, node_output)
```

Comment:
No longer necessary to check resource usage in the `skip_output` case because it'll be checked by the BlockBody of the node being rendered?

Reply:
The resource usage isn’t checked here in the “child node is empty” case now, an empty string is passed as the output buffer and thrown away without checking its size. But this seems fine since the “empty” node will run BlockBody logic itself and increment the render score. My question was aimed at confirming this theory.